src/database/network-table.c

```diff
@@ -327,7 +328,11 @@ bool unify_hwaddr(void)
 
 		// Obtain id and hwaddr of the most recent entry for this particular client
 		const int id = sqlite3_column_int(stmt, 0);
-		const char *hwaddr = (const char *)sqlite3_column_text(stmt, 1);
+		char *hwaddr = strdup((char*)sqlite3_column_text(stmt, 1));
+
+		// Finalize statement
+		sqlite3_finalize(stmt);
```

Comment:
Once the statement is finalized, isn't it undefined behavior to call `sqlite3_step` on it?
>  Any use of a prepared statement after it has been finalized can result in undefined and undesirable behavior such as segfaults and heap corruption.

https://www.sqlite.org/c3ref/finalize.html

Reply:
You are right, thank you for catching this. `sqlite3_reset()` should be sufficient here.